### PR TITLE
codacy-eslint rewrite and update to 6.7.2

### PR DESCRIPTION
**Pull request by lolgab**

### CODE REVIEW

File: src/parseTimeoutSeconds.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const defaultTimeout = 15 * 60
+
+export function parseTimeoutSeconds(timeoutString?: string): number {
+  if (!timeoutString) return defaultTimeout
+  let l = timeoutString.split(" ").filter(i => i) // remove empty strings
+  if (l.length != 2) return defaultTimeout
+  let number = parseInt(l[0])
+  let timeUnit = l[1]
+  if (isNaN(number)) return defaultTimeout
+  else if (timeUnit == "second" || timeUnit == "seconds") return number
```
Comment (ljmf00) [inline src/parseTimeoutSeconds.ts]: Can't we use a `switch` here?

File: src/model/Description.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  readonly parameters?: DescriptionParameter[]
+
+  constructor(patternId: string, title: string, description: string, timeToFix: number, parameters?: DescriptionParameter[]) {
+    this.patternId = patternId
+    this.title = title
+    this.description = description
+    this.timeToFix = timeToFix
+    this.parameters = parameters
+  }
+}
```
Comment (ljmf00) [inline src/model/Description.ts]: New line here for consistency

File: src/configCreator.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        result.baseConfig.extends = [] // TODO: Maintain base configurations without rules
+        result.baseConfig.overrides.extends = []
+        result.baseConfig.rules = patternsToRules(patterns)
+      }
+      result.useEslintrc = false
+      return result
+    }
+  }
+  let fileNames = await allFilesNames(srcDirPath)
+  if (fileNames.find(name => name.startsWith(".eslintrc")) != undefined)
```
Comment (ljmf00) [inline src/configCreator.ts]: Shouldn't be used strict equality `!==` here?

File: src/convertResults.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { flatMap } from "lodash"
+import { patternIdToCodacy } from "./model/Patterns"
+
+export function convertResults(report: CLIEngine.LintReport): CodacyResult[] {
+  return flatMap(report.results, result => {
+    let filename = result.filePath
+    return result.messages.map(m => {
+      let line = m.line
+      let message = m.message
+      let patternId = patternIdToCodacy(<string>m.ruleId) // TODO: Remove unsafe cast
```
Comment (ljmf00) [inline src/convertResults.ts]: Is it doable to remove `TODO` and fix this?

File: src/configCreator.ts (hunk 3 above)
Comment (lolgab) [reply src/configCreator.ts]: Done 👍

File: src/convertResults.ts (hunk 4 above)
Comment (lolgab) [reply src/convertResults.ts]: Done 👍

File: src/model/Description.ts (hunk 2 above)
Comment (lolgab) [reply src/model/Description.ts]: Done 👍

File: src/parseTimeoutSeconds.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const defaultTimeout = 15 * 60
+
+export function parseTimeoutSeconds(timeoutString?: string): number {
+  if (!timeoutString) return defaultTimeout
+  let l = timeoutString.split(" ").filter(i => i) // remove empty strings
```
[the remark this answers is not in the document]
Comment (lolgab) [reply src/parseTimeoutSeconds.ts]: Done 👍

File: src/parseTimeoutSeconds.ts (hunk 1 above)
Comment (lolgab) [reply src/parseTimeoutSeconds.ts]: Done 👍

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
+{
```
[the remark this answers is not in the document]
Comment (lolgab) [reply tsconfig.json]: Done 👍

File: src/fileUtils.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Codacyrc } from "./model/CodacyInput"
+import { readFile } from "./promisified"
+
+export async function readJsonFile(file: string): Promise<string | undefined> {
+  try {
+    return await readFile(file, "utf8")
```
Comment (peloycosta) [inline src/fileUtils.ts]: `return await` seems redundant just to catch the error. Could this go directly on index.ts?

File: src/fileUtils.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  } catch (e) {
+    console.error("Error reading .codacyrc file")
```
Comment (peloycosta) [inline src/fileUtils.ts]: Use `file` string here?

Comment (lolgab) [reply src/fileUtils.ts]: Done ✅

File: src/fileUtils.ts (hunk 7 above)
Comment (lolgab) [reply src/fileUtils.ts]: `index.ts` is already full of stuff. I prefer to leave it here.

**Pull request merged**